sktime/clustering/_time_series_k_means.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+"""Time Series K-Means Clusterer."""
+
+__author__ = ["Christopher Holder", "Tony Bagnall"]
+__all__ = ["TimeSeriesKMeans"]
+
+from typing import List
+from sktime.clustering.base.base_types import (
+    Metric_Parameter,
+    Data_Frame,
+    Numpy_Array,
+)
+from sktime.clustering.base.base import (
+    BaseCluster,
+    Init_Algo,
+    Averaging_Algo,
+    Averaging_Algo_Dict,
+)
+from sktime.clustering.partitioning._averaging_metrics import (
+    BarycenterAveraging,
+    MeanAveraging,
+)
+from sktime.clustering.partitioning._time_series_k_partition import TimeSeriesKPartition
+
+
+class TimeSeriesKMeans(TimeSeriesKPartition, BaseCluster):
+    """Time Series K-Means Clusterer.
+
+    This is a work in progress.
+    """
+
+    __averaging_algorithm_dict: Averaging_Algo_Dict = {
+        "dba": BarycenterAveraging,
+        "mean": MeanAveraging,
+    }
+
+    def __init__(
+        self,
+        n_clusters: int = 8,
+        init_algorithm: Init_Algo = "random",
+        max_iter: int = 300,
+        verbose: bool = False,
+        metric: Metric_Parameter = "dtw",
+        averaging_algorithm: Averaging_Algo = "mean",
+        averaging_algorithm_iterations: int = 50,
+    ):
+        """Constructor for TimeSeiresKMeans clusterer
+
+        Parameters
+        ----------
+            n_clusters: int, default = 8
```

Comment:
Additional indent here? 
```suggestion
        n_clusters: int, default = 8
```